go/libraries/doltcore/sqle/commit_diff_table.go

```diff
@@ -0,0 +1,397 @@
+		Type:     sql.Text,
+		Default:  defaultVal,
+		Nullable: false,
+		Source:   diffTblName,
+	})
+
+	return &CommitDiffTable{
+		name:        tblName,
+		dbName:      dbName,
+		ddb:         ddb,
+		workingRoot: workingRoot,
+		ss:          ss,
+		joiner:      j,
+		sqlSch:      sqlSch,
+	}, nil
+}
+
+func calcSuperDuperSchema(ctx context.Context, ddb *doltdb.DoltDB, working *doltdb.RootValue, tblName string) (*schema.SuperSchema, error) {
+	refs, err := ddb.GetRefs(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var superSchemas []*schema.SuperSchema
+	ss, found, err := working.GetSuperSchema(ctx, tblName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if found {
+		superSchemas = append(superSchemas, ss)
+	}
+
+	for _, ref := range refs {
+		cm, err := ddb.ResolveRef(ctx, ref)
+
+		if err != nil {
+			return nil, err
+		}
+
+		cmRoot, err := cm.GetRootValue()
+
+		if err != nil {
+			return nil, err
+		}
+
+		ss, found, err = cmRoot.GetSuperSchema(ctx, tblName)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if found {
+			superSchemas = append(superSchemas, ss)
+		}
+	}
+
+	if len(superSchemas) == 0 {
+		return nil, sql.ErrTableNotFound.New(tblName)
+	}
+
+	superDuperSchema, err := schema.SuperSchemaUnion(superSchemas...)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return superDuperSchema, nil
+}
+
+func (dt *CommitDiffTable) Name() string {
+	return doltdb.DoltCommitDiffTablePrefix + dt.name
+}
+
+func (dt *CommitDiffTable) String() string {
+	return doltdb.DoltCommitDiffTablePrefix + dt.name
+}
+
+func (dt *CommitDiffTable) Schema() sql.Schema {
+	return dt.sqlSch
+}
+
+type SliceOfPartitionsItr struct {
+	partitions []sql.Partition
+	i          int
+	mu         *sync.Mutex
+}
+
+func NewSliceOfPartitionsItr(partitions []sql.Partition) *SliceOfPartitionsItr {
+	return &SliceOfPartitionsItr{
+		partitions: partitions,
+		mu:         &sync.Mutex{},
+	}
+}
+
+func (itr *SliceOfPartitionsItr) Next() (sql.Partition, error) {
+	itr.mu.Lock()
+	defer itr.mu.Unlock()
+
+	if itr.i >= len(itr.partitions) {
+		return nil, io.EOF
+	}
+
+	next := itr.partitions[itr.i]
+	itr.i++
+
+	return next, nil
+}
+
+func (itr *SliceOfPartitionsItr) Close() error {
+	return nil
+}
+
+func (dt *CommitDiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
+	if dt.requiredFilterErr != nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), dt.requiredFilterErr)
+	} else if dt.toCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneToCommit)
+	} else if dt.fromCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneFromCommit)
+	}
+
+	toRoot, toName, toDate, err := dt.rootValForFilter(ctx, dt.toCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromRoot, fromName, fromDate, err := dt.rootValForFilter(ctx, dt.fromCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toTable, _, err := toRoot.GetTable(ctx, dt.name)
```

Comment:
Probably want to handle `ok` on these, this will panic if the lookup fails otherwise

Reply:
it doesn't panic.  It handles nils as a table not existing and all rows being added.  I updated my bats test to verify.